### PR TITLE
Update autosave

### PR DESCRIPTION
**Pull request by jon-stevens**
This update allows the consumer to optionally provide data to the component rather than have it retrieve it from an endpoint.

### CODE REVIEW

File: packages/global-autocomplete/README.md [hunk 1]
```diff
@@ -39,7 +39,8 @@ const args = {
     selector: '[data-component-autocomplete]',
     onSelect: onSelect,
     searchError: onError,
-    endPoint: 'autocomplete?q=',
+    endPoint: 'autocomplete?q=',  // OPTIONAL: Provide an endpoint if you want the component to request data
```
Comment (jpw) [inline packages/global-autocomplete/README.md]: `endpoint`

File: packages/global-autocomplete/js/index.js [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
+		minChars = 0,
+		componentName,
+		data,
+		onSelect,
+		inputDelay = 300,
+		requestTimeout = 2000,
+		headers = {},
+		searchError,
+		resultsCallBack
+	} = args;
```
Comment (jpw) [inline packages/global-autocomplete/js/index.js]: ohhhh that's a nice pattern, i will have to steal that 🙂

File: packages/global-autocomplete/js/index.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+	const input = document.querySelector(selector);
+
+	if (!endPoint && !data) {
+		console.error('Please provide data or a data endpoint');
+		return;
+	}
+
 	const container = () => {
-		return document.querySelector(`.${args.resultsContainerSelector}`);
+		return document.querySelector(`.${resultsContainerSelector}`);
```
Comment (jpw) [inline packages/global-autocomplete/js/index.js]: Should it be a restriction that `resultsContainerSelector` has to be a class? Docs imply not...

File: packages/global-autocomplete/js/index.js [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
 	};
 	const suggestions = () => {
-		return Array.from(document.querySelectorAll(`.${args.resultSelector}`));
+		return Array.from(document.querySelectorAll(`.${resultSelector}`));
```
Comment (jpw) [inline packages/global-autocomplete/js/index.js]: ditto

File: packages/global-autocomplete/js/index.js [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
+		data,
+		onSelect,
+		inputDelay = 300,
+		requestTimeout = 2000,
+		headers = {},
+		searchError,
+		resultsCallBack
+	} = args;
+
+	const input = document.querySelector(selector);
```
Comment (jpw) [inline packages/global-autocomplete/js/index.js]: Is `selector` a required arg?

File: packages/global-autocomplete/js/index.js [hunk 6]
```diff
@@ -122,9 +135,14 @@ const autoComplete = args => {
 		}
 	};
 
-	const sendQuery = term => {
-		let getSuggestions = new Promise((resolve, reject) => {
-			fetch(endpoint + term, {
+	const handleData = term => {
+		if (data) {
```
Comment (jpw) [inline packages/global-autocomplete/js/index.js]: A pet peeve of mine is `data` as a variable name. 

1. By the time I got to this point in the code, I had already forgotten what `data` referred to.
1. In `generateSuggestions` above, you also use `data` locally (which is fine there IMO as it's dealing with fetched _and_ supplied result data, so it's agnosticism as to the source of the data makes sense), whereas here you are referring to the `data` argument... Maybe a more descriptive name for the argument? `resultData`? `resultDataArg`? `staticResultData`? `suppliedResultData`?

File: packages/global-autocomplete/README.md [hunk 7]
```diff
@@ -39,7 +39,8 @@ const args = {
...
+    data:                   // OPTIONAL: Provide JSON if you want to fetch data and transform it yourself.
```
Comment (jpw) [inline packages/global-autocomplete/README.md]: If I am reading the code correctly, `data` takes precedence over `endpoint` if both are supplied..? maybe that should be mentioned here?

Comment (jon-stevens) [reply packages/global-autocomplete/README.md]: ah yes, good point

File: packages/global-autocomplete/js/index.js (hunk 6 above)
Comment (jon-stevens) [reply packages/global-autocomplete/js/index.js]: yeah good point

File: packages/global-autocomplete/js/index.js (hunk 5 above)
Comment (jon-stevens) [reply packages/global-autocomplete/js/index.js]: Yep. I have now added a check

File: packages/global-autocomplete/js/index.js (hunk 2 above)
Comment (jon-stevens) [reply packages/global-autocomplete/js/index.js]: 👍

**Pull request merged**